l10n_it_fatturapa_out_rc/tests/test_e_invoice_out_rc.py

```diff
@@ -240,3 +241,64 @@ def test_extra_EU(self):
         xml_content = base64.decodebytes(attachment.datas)
         self.check_content(
             xml_content, 'IT10538570960_00004.xml', "l10n_it_fatturapa_out_rc")
+
+    def _configure_be_supplier(self, supplier):
+        supplier_form = Form(supplier)
+        supplier_form.property_payment_term_id = self.term_15_30
+        supplier_form.vat = "BE0477472701"
+        supplier_form.street = "Street"
+        supplier_form.zip = "12345"
+        supplier_form.city = "city"
+        supplier_form.country_id = self.env.ref("base.be")
+        supplier = supplier_form.save()
+        return supplier
+
+    def _create_rc_bill(self, supplier):
+        bill_model = self.invoice_model.with_context(type='in_invoice')
+        bill_form = Form(bill_model)
+        bill_form.partner_id = supplier
+        bill_form.date_invoice = '2020-12-01'
+        with bill_form.invoice_line_ids.new() as line:
+            line.name = 'Bill for sample product'
+            line.product_id = self.sample_product
+            line.invoice_line_tax_ids.clear()
+            line.invoice_line_tax_ids.add(self.tax_22ai)
+        bill = bill_form.save()
+        bill.action_invoice_open()
+        return bill
+
+    def test_intra_EU_multiple_suppliers(self):
+        """
+        Export multiple self-invoices coming from different suppliers.
+        """
+        # Arrange: Create 2 self invoices from 2 supplier bills
+        # having different suppliers
+        be_supplier = self._configure_be_supplier(self.supplier_intraEU)
+        be_bill = self._create_rc_bill(be_supplier)
+        be_self_invoice = be_bill.rc_self_invoice_id
+        self.assertTrue(be_self_invoice)
+
+        bg_supplier = be_supplier.copy(default={
+            'vat': 'BG1234567892',
+            'country_id': self.env.ref("base.bg").id,
+        })
+        bg_bill = be_bill.copy(default={
+            'partner_id': bg_supplier.id,
+        })
+        bg_bill.action_invoice_open()
+        bg_self_invoice = bg_bill.rc_self_invoice_id
```

Comment:
Nella configurazione con autofattura passiva, per la fattura fornitore `rc_self_invoice_id` è vuoto.
È invece valorizzato `rc_self_purchase_invoice_id` che, a sua volta, avrà  `rc_self_invoice_id` (la fattura da mandare a SDI)

Reply:
Ho aggiunto un test per coprire il caso in cui il tipo inversione contabile ha "Con autofattura passiva aggiuntiva" abilitato, puoi controllare se ora va meglio?